src/packagedcode/cargo.py

```diff
@@ -70,33 +71,46 @@ def get_package_root(cls, manifest_resource, codebase):
         return manifest_resource.parent(codebase)
 
     def repository_homepage_url(self, baseurl=default_web_baseurl):
-        return '{}/crates/{}'.format(baseurl, self.name)
+        if self.name:
+            return '{}/crates/{}'.format(baseurl, self.name)
+        return None
 
     def repository_download_url(self, baseurl=default_download_baseurl):
-        return '{}/crates/{}/{}/download'.format(baseurl, self.name, self.version)
+        if self.name and self.version:
+            return '{}/crates/{}/{}/download'.format(baseurl, self.name, self.version)
+        return None
 
     def api_data_url(self, baseurl=default_api_baseurl):
-        return '{}/crates/{}'.format(baseurl, self.name)
+        if self.name:
+            return '{}/crates/{}'.format(baseurl, self.name)
+        return None
```

Comment:
I would prefer if we followed this convention:
```
if not self.name:
    return None
return '{}/crates/{}'.format()
```
IMO this is easier to read and follow.

Reply:
Okay. I will change it soon.